src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2889 @@
+
+
+  REAL(r_kind) :: n0rain,n0snow,n0hail,n0grpl,rhosnow,rhohail,rhogrpl,   &
+          alpharain,alphasnow,alphagrpl,alphahail
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  N0r=n0rain
+  N0s=n0snow
+  N0h=n0hail
+  N0g=n0grpl
+
+  rhos=rhosnow
+  rhoh=rhohail
+  rhog=rhogrpl
+
+  alphar = alpharain
+  alphas = alphasnow
+  alphag = alphagrpl
+  alphah = alphahail
+
+  END SUBROUTINE model_dsd
+
+  SUBROUTINE coeff_hail(fw,qml)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! Scattering matrix coefficient for hail
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/27/2007
+!
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Define variables:
+!-----------------------------------------------------------------------
+
+  REAL(r_kind), INTENT(IN) :: fw, qml
+
+! DTD: test 02/16/2012, do not adjust coefficient downward for small hail
+!  IF(qml < q_threshold) THEN
+!     sf = 4*qml*1.e3
+!  ELSE
+     sf = 0.8
+!  ENDIF
+
+  sigmah=pi/3*(1-sf*fw)
+  Ah=.125*(3+4*exp(-2*sigmah**2)+exp(-8*sigmah**2))
+  Bh=.125*(3-4*exp(-2*sigmah**2)+exp(-8*sigmah**2))
+  Ch=.125*(1-exp(-8*sigmah**2))
+  Dh=.125*(3+exp(-8*sigmah**2))
+  Ckh=exp(-2*sigmah**2)
+
+  END SUBROUTINE coeff_hail
+
+SUBROUTINE coeff_grpl(fw,qml)
+
+!-----------------------------------------------------------------------
+!
+! PURPOSE:
+!
+! Scattering matrix coefficient for graupel
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, 1/27/2007
+!
+! MODIFIED: Dan Dawson, 02/16/2012
+!           Made separate version for graupel.
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Define variables:
+!-----------------------------------------------------------------------
+
+  REAL(r_kind), INTENT(IN) :: fw, qml
+
+! DTD: test 02/16/2012, do not adjust coefficient downward for small hail
+!  IF(qml < q_threshold) THEN
+!     sf = 4*qml*1.e3
+!  ELSE
+     sf = 0.8
+!  ENDIF
+
+  sigmag=pi/3*(1-sf*fw)
+  Ag=.125*(3+4*exp(-2*sigmag**2)+exp(-8*sigmag**2))
+  Bg=.125*(3-4*exp(-2*sigmag**2)+exp(-8*sigmag**2))
+  Cg=.125*(1-exp(-8*sigmag**2))
+  Dg=.125*(3+exp(-8*sigmag**2))
+  Ckg=exp(-2*sigmag**2)
+
+  END SUBROUTINE coeff_grpl
+
+  TYPE(T_obs_dual) FUNCTION assign_Refl(var1,var2,var3,var4)
+       REAL(r_kind) :: var1,var2,var3,var4,var5
```

Comment:
The variable, var5, has been identified as unused.  Please remove unused variable.

Reply:
We deleted unnecessary variable as you commented.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R662